### ISSUE TITLE
Capybara 2.15.3 visit to site url port uses port from app_host or server port

### ISSUE THREAD

**Issue by BrianHawley** [teamcapybara/capybara#1921]
## Meta
Capybara Version:
2.15.3
Driver Information (and browser if relevant):
selenium-webdriver 3.6.0 (not relevant)

## Expected Behavior
Set `Capybara.app_host` to a url with a port and `Capybara.run_server = false`.
or
Leave `Capybara.run_server = true` and set `Capybara.always_include_port = true`.

Visit another host, using a url with no port specified.
You should go to that host.

## Actual Behavior
You go to the host with the port from `app_host` or the server port, respectively.

## Steps to reproduce
```ruby
Capybara.app_host = 'http://localhost:5000'
Capybara.run_server = false
Capybara.current_session.visit('https://www.facebook.com')
```

or

```ruby
Capybara.always_include_port = true
Capybara.current_session.visit('https://www.facebook.com')
```

This happened as a side effect of the #1918 changes, and it doesn't happen in earlier versions.

**Comment by BrianHawley** [comment]
This would probably lead to basic auth credentials being forwarded to external sites too, if basic auth is specified in the `app_host`.

**Comment by twalpole** [comment]
Hmmm -- with a server and `always_include_port = true` then it should visit 'https://www.facebook.com:<server_port>`  - that's the whole point of `always_include_port` and should have previously.  The fix just included should have made it that you could specify `https://www.facebook.com:443` to prevent the port overwrite.  The case where `run_server = false` does seem like a regression.

**Comment by twalpole** [comment]
@BrianHawley  Please try branch issue_1921 and see if the first part behaves as expected -- the second half of this is expected behavior (when always_include_port is true the port should be set unless explicitly provided in the visit url)

**Comment by twalpole** [comment]
Fixed in 2.15.4

**Comment by BrianHawley** [comment]
@twalpole sorry for the delay in replying. After manual checking of 2.15.4, in all the cases I noticed and the ones you mentioned above, it's fixed.
